dispatcher/MongoConnect.py

```diff
@@ -504,8 +504,17 @@ def set_stop_time(self, number, detectors, force):
                                 'max': {'$max': '$rate'}}}
                     ]):
                     rate[doc['_id']] = {'avg': doc['avg'], 'max': doc['max']}
+                channels = set()
+                if 'tpc' in detectors:
+                    # figure out which channels weren't running
+                    readers = list(self.latest_status[detectors]['readers'].keys())
+                    for doc in self.collections['node_status'].find({'host': {'$in': readers}, 'number': int(number)}):
+                        channels |= set(map(int, doc['channels'].keys()))
+                updates = {'rate': rate}
+                if len(channels):
```

Comment:
Would you want to also store it if the list is empty? Might make querying later a little easier.

Reply:
I think an empty list behaves identically to a non-existent list. I don't really see this field getting extensively queried.